_posts/en/posts/2018-09-20-notice.md

```diff
@@ -29,7 +29,19 @@ However, it still remains critical that affected users upgrade and apply the lat
 Technical Details
 =================
 
-In Bitcoin Core 0.14, an optimization was added (Bitcoin Core PR #9049) which avoided a costly check during initial pre-relay block validation that multiple inputs within a single transaction did not spend the same input twice which was added in 2012 (PR #443). While the UTXO-updating logic has sufficient knowledge to check that such a condition is not violated in 0.14 it only did so in a sanity check assertion and not with full error handling (it did, however, fully handle this case twice in prior to 0.8).
+The original implementation of Bitcoin ensured that no transaction
+within a block spent an input that had already been spent on the same
+block chain.  However, it failed to ensure that the transactions it
+relayed and accepted into its memory pool only spent the same input
+once, so PR #443 was opened to add a rule that rejected transactions
+with duplicate inputs.  This became part of the 0.4 release of Bitcoin.
+
+This new rule was added to the function used to validate all
+transactions, whether they were unconfirmed transactions being relayed
+or transactions included in blocks, which meant that Bitcoin software
+was redundantly checking blocks for duplicate inputs twice.
```

Comment:
I think this could be made even clearer with something like:

```
redundantly checking blocks for duplicate inputs twice:
- once when validating the transaction
- once when updating the set of unspend outputs (UTXOs)
```
That would make the following section about 'UTXO-updating logic' clearer.